relay-server/src/actors/envelopes.rs

```diff
@@ -634,13 +630,6 @@ impl Handler<HandleEnvelope> for EnvelopeManager {
                 } else {
                     relay_log::debug!("dropped envelope: {}", LogError(&error));
                 }
-
-                if let ProcessingError::Timeout = error {
-                    // handle the last failure (the timeout)
-                    if let Some(outcome) = outcome {
-                        envelope_context.borrow().send_outcomes(outcome);
-                    }
-                }
```

Comment:
it's kind of sad to be able to use the power of exhaustive matching here.  but that's not this PRs fault so just an aside